### PR TITLE
Make group.id Kafka input config field optional

### PR DESCRIPTION
**Pull request by Karakatiza666**
Is this a user-visible change (yes/no): yes

Commit 1:
Fix https://github.com/feldera/feldera/issues/1021

Commit 2: 
Add ability to edit connector configuration JSON directly: Fix https://github.com/feldera/feldera/issues/1132

### CODE REVIEW

**Comment by Karakatiza666** [comment]
image
image

**Comment by gz** [comment]
unknown error when adding connector:
https://github.com/feldera/feldera/assets/127654/1afc1ceb-23c2-40fb-8a07-298d078e9e4f

**Comment by gz** [comment]
two bugs:

- the cursor (unexpectedly) jumps to the end of the config when trying to edit
- when pressing update after modification nothing happens

https://github.com/feldera/feldera/assets/127654/9a459b04-7dbe-4654-ad8f-206958c47623

**Comment by gz** [comment]
unknown config properties are silently removed (IMO it's be better to keep them if the user writes it explicitly as it could be that there are options the UI doesn't support yet but this would still allow an user to add them):

https://github.com/feldera/feldera/assets/127654/fc4fcf0c-4758-4ab8-bf12-29f0efd9d4fc

**Comment by gz** [comment]
maybe ignore the previous comment, it seems the bug is it doesn't update the config in the first place?

https://github.com/feldera/feldera/assets/127654/cedbf16f-0b1d-4557-8a32-b130fae02a2f

**Comment by gz** [comment]
I think this is a good opportunity to write use the test framework and write tests for these things

**Comment by gz** [comment]
image

I'd call this "Edit JSON"

File: web-console/src/lib/components/connectors/dialogs/GenericEditorConnector.tsx [hunk 1]
```diff
@@ -52,28 +45,24 @@ export const ConfigEditorDialog = (props: ConnectorDialogProps) => {
...
-      {
-        transport: {
-          name: 'transport-name',
-          config: {
-            property: 'value'
-          }
-        },
-        format: {
-          name: 'csv'
+    config: {
```
Comment (mihaibudiu) [inline web-console/src/lib/components/connectors/dialogs/GenericEditorConnector.tsx:48]: stronger typing is always good

File: web-console/src/lib/components/connectors/dialogs/tabs/GenericConnectorForm.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+  function handleEditorDidMount(editor: editor.IStandaloneCodeEditor, monaco: Monaco) {
+    editorRef.current = editor
+    // Only process input when we have finished typing
+    editor.onDidBlurEditorText(() => {
+      try {
+        const v = props.configFromText(editor.getValue() ?? '')
+        ctx.setValue('config', v)
+        monaco.editor.setModelMarkers(editorRef.current!.getModel()!, 'parse-errors', [])
+      } catch (e) {
+        invariant(e instanceof Error)
```
Comment (mihaibudiu) [inline web-console/src/lib/components/connectors/dialogs/tabs/GenericConnectorForm.tsx:116]: so there are no libraries doing this already?

**Review by mihaibudiu — APPROVED** [review]

**Comment by mihaibudiu** [comment]
Looks to me like this PR contains also other refactorings besides the PR title.

File: web-console/src/lib/components/connectors/dialogs/tabs/GenericConnectorForm.tsx (hunk 2 above)
Comment (Karakatiza666) [reply web-console/src/lib/components/connectors/dialogs/tabs/GenericConnectorForm.tsx:116]: For generating error markers from errors? Doesn't seem to be. I will create helper functions when there are more instances of similar code to generalize

**Pull request merged**